File: leo2-central/src/main/java/org/deku/leo2/central/entities/DepotJooqRepository.java
package org.deku.leo2.central.entities;

import org.deku.leo2.central.entities.jooq.Tables;
import org.deku.leo2.central.entities.jooq.tables.records.TbldepotlisteRecord;
import org.jooq.DSLContext;
import org.springframework.transaction.annotation.Transactional;

import javax.inject.Inject;
import javax.inject.Named;
import java.util.List;

/**
 * Created by masc on 07.05.15.
 */
@Named
public class DepotJooqRepository {
    @Inject
    DSLContext mDSLContext;

    @Transactional("jooq")
    public List<TbldepotlisteRecord> findAll() {
        return mDSLContext.select()
                .from(Tables.TBLDEPOTLISTE)
                .fetchInto(TbldepotlisteRecord.class);
    }
}

File: leoz-central/src/main/java/org/deku/leoz/central/data/repositories/DepotJooqRepository.kt
package org.deku.leoz.central.data.repositories

import org.deku.leoz.central.config.PersistenceConfiguration
import org.deku.leoz.central.data.entities.jooq.Tables
import org.deku.leoz.central.data.entities.jooq.tables.records.TbldepotlisteRecord
import org.jooq.DSLContext
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.transaction.annotation.Transactional

import javax.inject.Inject
import javax.inject.Named

/**
 * Created by masc on 07.05.15.
 */
@Named
class DepotJooqRepository {
    @Inject
    @Qualifier(PersistenceConfiguration.QUALIFIER)
    private lateinit var dslContext: DSLContext

    @Transactional(PersistenceConfiguration.QUALIFIER)
    fun findAll(): List<TbldepotlisteRecord> {
        return dslContext
                .select()
                .from(Tables.TBLDEPOTLISTE)
                .fetchInto(TbldepotlisteRecord::class.java)
    }
}
